spring-boot/admin/src/main/java/eu/coatrack/admin/service/OAuthUserDetailsService.java

```diff
@@ -56,19 +73,29 @@ private OAuth2User getLoggedInUser() {
     }
 
     public String getLoginNameFromLoggedInUser() {
-        return getLoggedInUser().getAttribute("login");
+        return getLoggedInUser().getAttribute(attributeNameForLogin);
     }
 
     public String getNameFromLoggedInUser() {
-        return getLoggedInUser().getAttribute("name");
+        return getLoggedInUser().getAttribute(attributeNameForName);
     }
 
     public String getCompanyFromLoggedInUser() {
-        return getLoggedInUser().getAttribute("company");
+        if ((attributeNameForCompany).equals(KEYROCK_COMPANY_ATTRIBUTE)) {
+            return getFirstOrganizationFromKeyrock();
+        } else
+            return getLoggedInUser().getAttribute(attributeNameForCompany);
```

Comment:
You do this call also in the method `getFirstOrganizationFromKeyrock`, which means that this is on another abstraction level. Maybe harmonize?

```suggestion
            return getLoggedInUser().getAttribute(attributeNameForCompany);
```

Reply:
If the changes on the below thread are dropped, then this thread will be resolved

Reply:
Done